packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts

```diff
@@ -28,12 +29,26 @@ export const useGridKeyboardNavigation = (
   const logger = useGridLogger(apiRef, 'useGridKeyboardNavigation');
   const currentPage = useGridVisibleRows(apiRef, props);
 
+  /**
+   * @param {number} colIndex Index of the column to focus
+   * @param {number} rowIndex index of the row to focus
+   * @param {string} closestColumnToUse Which closest column cell to use when the cell is spanned by `colSpan`.
+   */
   const goToCell = React.useCallback(
-    (colIndex: number, rowIndex: number) => {
+    (colIndex: number, rowIndex: number, closestColumnToUse: 'left' | 'right' = 'left') => {
+      const visibleSortedRows = gridVisibleSortedRowEntriesSelector(apiRef);
+      const rowId = visibleSortedRows[rowIndex]?.id;
+      const nextCellColSpanInfo = apiRef.current.unstable_getCellColSpanInfo(rowId, colIndex);
+      if (nextCellColSpanInfo && nextCellColSpanInfo.spannedByColSpan) {
+        if (closestColumnToUse === 'left') {
+          colIndex = nextCellColSpanInfo.leftVisibleCellIndex;
+        } else if (closestColumnToUse === 'right') {
+          colIndex = nextCellColSpanInfo.rightVisibleCellIndex;
+        }
+      }
       logger.debug(`Navigating to cell row ${rowIndex}, col ${colIndex}`);
       apiRef.current.scrollToIndexes({ colIndex, rowIndex });
       const field = apiRef.current.getVisibleColumns()[colIndex].field;
-      const visibleSortedRows = gridVisibleSortedRowEntriesSelector(apiRef);
       const node = visibleSortedRows[rowIndex];
```

Comment:
You already have `visibleSortedRows[rowIndex]` above.

Reply:
I was referring to https://github.com/mui/mui-x/pull/4020/files#diff-627fe964accb535cecc2456eab54c357d5a7fb937175a164a5aa16a115a4c5bbR40

It's nothing really important.

Reply:
Right, I can just reuse `rowId` and remove this line

```js
const node = visibleSortedRows[rowIndex]
```